aya-obj/src/btf/relocation.rs

```diff
@@ -862,9 +878,30 @@ impl ComputedRelocation {
                     relocation_number: rel.number,
                 })?;
 
+        if self.target.is_none() {
+            let is_ld_imm64 = ins.code == (BPF_LD | BPF_DW) as u8;
+
+            poison_insn(ins);
+
+            if is_ld_imm64 {
+                let next_ins = instructions.get_mut(ins_index + 1).ok_or(
+                    RelocationError::InvalidInstructionIndex {
+                        index: (ins_index + 1) * mem::size_of::<bpf_insn>(),
+                        num_instructions,
+                        relocation_number: rel.number,
+                    },
+                )?;
+
+                poison_insn(next_ins);
+            }
+
+            return Ok(());
+        }
+
         let class = (ins.code & 0x07) as u32;
 
-        let target_value = self.target.value;
+        let target = self.target.as_ref().unwrap();
```

Comment:
instead of doing if is_none() above and then unwrap, use a match statement and
return early in the None case

Reply:
Also, would be good having an integration test for this